### PR TITLE
phpstan level 6 fixes

### PR DESCRIPTION
**Pull request by acrobat**
Added missing docblocks to prepare code for typehints/return types + fixed some incorrect types in parameters

### CODE REVIEW

File: lib/Github/Exception/ApiLimitExceedException.php [hunk 1]
```diff
@@ -9,9 +9,17 @@
...
+    /** @var int */
     private $limit;
+    /** @var int */
     private $reset;
 
+    /**
+     * @param int             $limit
+     * @param int             $reset
+     * @param int             $code
+     * @param \Throwable|null $previous
```
Comment (GrahamCampbell) [inline lib/Github/Exception/ApiLimitExceedException.php:21]: Doesn't 2.x support PHP 5 still? Is previous actually always an Exception? Can we just typehint `\Exception|null` here?

File: lib/Github/Exception/MissingArgumentException.php [hunk 2]
```diff
@@ -9,6 +9,11 @@
  */
 class MissingArgumentException extends ErrorException
 {
+    /**
+     * @param string|array    $required
+     * @param int             $code
+     * @param \Throwable|null $previous
```
Comment (GrahamCampbell) [inline lib/Github/Exception/MissingArgumentException.php:15]: Same

File: lib/Github/Exception/TwoFactorAuthenticationRequiredException.php [hunk 3]
```diff
@@ -4,14 +4,23 @@
 
 class TwoFactorAuthenticationRequiredException extends RuntimeException
 {
+    /** @var string */
     private $type;
 
+    /**
+     * @param string          $type
+     * @param int             $code
+     * @param \Throwable|null $previous
```
Comment (GrahamCampbell) [inline lib/Github/Exception/TwoFactorAuthenticationRequiredException.php:13]: same

File: lib/Github/HttpClient/Message/ResponseMediator.php [hunk 4]
```diff
@@ -59,7 +59,7 @@ public static function getApiLimit(ResponseInterface $response)
         $remainingCalls = self::getHeader($response, 'X-RateLimit-Remaining');
 
         if (null !== $remainingCalls && 1 > $remainingCalls) {
-            throw new ApiLimitExceedException($remainingCalls);
+            throw new ApiLimitExceedException((int) $remainingCalls);
```
Comment (GrahamCampbell) [inline lib/Github/HttpClient/Message/ResponseMediator.php:62]: Shouldn't this cast happen before we compare with `1`?

File: lib/Github/Exception/ApiLimitExceedException.php (hunk 1 above)
Comment (acrobat) [reply lib/Github/Exception/ApiLimitExceedException.php:21]: No, I've bumped php a while ago. We support 7.1+

https://github.com/KnpLabs/php-github-api/blob/6a80368db3d9f742df0c109a27a2d9227b1800d5/composer.json#L20

File: lib/Github/HttpClient/Message/ResponseMediator.php (hunk 4 above)
Comment (acrobat) [reply lib/Github/HttpClient/Message/ResponseMediator.php:62]: Yes you are right, what should we do with the current return type of this method? It has `@return null|string` change it to `int` or return the original result of the getHeader call and keep the current return type?

Comment (GrahamCampbell) [reply lib/Github/HttpClient/Message/ResponseMediator.php:62]: Shame PHP never added the `(?int)` casting I suggested. Would have been perfect for line 61. ;)

Comment (GrahamCampbell) [reply lib/Github/HttpClient/Message/ResponseMediator.php:62]: Though, casting null to zero would be fine. We could then fix the return to change zero back to null. Give me a sec...

Comment (GrahamCampbell) [reply lib/Github/HttpClient/Message/ResponseMediator.php:62]: https://github.com/acrobat/php-github-api/pull/1

File: lib/Github/HttpClient/Message/ResponseMediator.php [hunk 5]
```diff
@@ -65,10 +65,10 @@ public static function getApiLimit(ResponseInterface $response)
         $remainingCalls = (int) $remainingCallsHeader;
 
         if (1 > $remainingCalls) {
-            throw new ApiLimitExceedException((int) $remainingCalls);
+            throw new ApiLimitExceedException($remainingCalls);
         }
 
-        return $remainingCalls;
+        return $remainingCallsHeader;
```
Comment (GrahamCampbell) [inline lib/Github/HttpClient/Message/ResponseMediator.php:71]: Whoops. Maybe in 3.x this function's return type can be changed to int|null.

Comment (acrobat) [reply lib/Github/HttpClient/Message/ResponseMediator.php:71]: We can prepare it in 2.x already! Maybe something like?

```php
public static function getApiLimit(ResponseInterface $response, $returnInt = false)
    {
        if (false === $returnInt) {
            @trigger_error('Not passing "true" as value for "$returnInt" is deprecated. In v3 this function will return "int|null".', E_USER_DEPRECATED);
        }

        $remainingCallsHeader = self::getHeader($response, 'X-RateLimit-Remaining');

        if (null === $remainingCallsHeader) {
            return null;
        }

        $remainingCalls = (int) $remainingCallsHeader;

        if (1 > $remainingCalls) {
            throw new ApiLimitExceedException($remainingCalls);
        }

        if ($returnInt) {
            return $remainingCalls;
        }

        return $remainingCallsHeader;
    }
```

Comment (GrahamCampbell) [reply lib/Github/HttpClient/Message/ResponseMediator.php:71]: Maybe is overkill. ;)

Comment (acrobat) [reply lib/Github/HttpClient/Message/ResponseMediator.php:71]: Maybe you are right, let's leave it for now and see how we do it in future releases!

**Pull request merged**